### PR TITLE
Close CSS Code Hints after inputting property value

### PR DESCRIPTION
**Pull request by marcelgerber**
For #9496 and #4888

### CODE REVIEW

File: src/extensions/default/CSSCodeHints/main.js [hunk 1]
```diff
@@ -321,7 +321,13 @@ define(function (require, exports, module) {
                 selectInitial = true;
             }
             
-            lastContext = CSSUtils.PROP_NAME;
+            if (lastContext === CSSUtils.PROP_VALUE) {
+                // close the session if we're coming from a property value
+                // see https://github.com/adobe/brackets/issues/9496
+                return false;
```
Comment (nethip) [inline src/extensions/default/CSSCodeHints/main.js]: @MarcelGerber Just a small thing. To adhere to how it is done everywhere can you change this to `return NULL`.

**Comment by nethip** [comment]
@MarcelGerber I took this change and it does not seem to solve #9496. I am not sure if this makes the issue better though.

File: src/extensions/default/CSSCodeHints/main.js (hunk 1 above)
Comment (MarcelGerber) [inline src/extensions/default/CSSCodeHints/main.js]: Fixed (& rebased)

**Comment by MarcelGerber** [comment]
Hm, it definitely works for me. So, just to be sure, these are the steps I followed:
1. Type `position: absolute;`
2. Place your mouse before the `p`
3. Type (without using Code Hints!) `left: auto;` - without a trailing space
4. Press Enter

Before: `left: auto;align-content: : absolute;`
After: `left: auto;[Newline]position: absolute;`

**Comment by kylelondonuk** [comment]
Hi, I opened issue #9496 and was wondering whether this fix is available yet? the problem still seems to exist in Brackets 1.3 downloaded today.

Cheers

**Comment by nethip** [comment]
@kylelondonuk This is not merged yet.

**Comment by ficristo** [comment]
FWIW I can confirm the [STR](https://github.com/adobe/brackets/pull/10524#issuecomment-73233812) above and this PR fixes it.

**Comment by MarcelGerber** [comment]
I do think this fix should be reviewed soon. Would you mind? Maybe @abose?

**Comment by ficristo** [comment]
There is a test failing under `Extensions \ CSS Code Hinting`. Could you take a look?

By the way I'd write the code like the following: I find it easier to read.
```diff
-            lastContext = CSSUtils.PROP_NAME;
+            if (lastContext === CSSUtils.PROP_VALUE) {
+                // Close the session if we're coming from a property value.
+                // See https://github.com/adobe/brackets/issues/9496
+                return null;
+            }
+
+            lastContext = CSSUtils.PROP_NAME;
             needle = needle.substr(0, this.info.offset);
```

**Comment by MarcelGerber** [comment]
@ficristo With latest master merged in, all tests pass for me.
I've also incorporated your suggested code style change.

**Comment by ficristo** [comment]
You are missing a brace.

**Comment by MarcelGerber** [comment]
Argh, yesterday simply wasn't my day. Fixed it now and squashed the commits as.
Should be good to go after one last look.

**Comment by ficristo** [comment]
Thank you, LGTM.

**Pull request merged**